doc/sphinx-guides/source/developers/api.rst

```diff
@@ -0,0 +1,115 @@
+            .internalError(ex)
+            // Add the complete URL & HTTP method the client tried to use
+            .request(request)
+            // Write all this to the logs, including the stack trace from the exception
+            .log(logger, Level.SEVERE, Optional.of(ex))
+            // Build the JAX-RS response. Done!
+            .build();
+
+Exception Handling
+------------------
+
+We have to differ between three types of exceptions thrown during API usage:
+
+1. Servlet exceptions, based on `javax.servlet.ServletException <https://jakarta.ee/specifications/platform/8/apidocs/javax/servlet/ServletException.html>`_
+2. JAX-RS-native exceptions, based on `javax.ws.rs.WebApplicationException <https://jakarta.ee/specifications/platform/8/apidocs/javax/ws/rs/WebApplicationException.html>`_
+3. Any other Java exception
+
+When accessing the API, requests are handled by the servlet container before handing over to Jersey and from there to
+the lower layers of bean validation, database connections and all other stuff you might use through the API:
+
+.. graphviz::
+
+    digraph G {
+      rankdir = LR
+
+      node [shape=plain] Client;
+      node [shape=egg] Network;
+
+      subgraph cluster_0 {
+          node [shape=box,label="Filters, ..."] F;
+          node [shape=box,label="JAX-RS/Jersey"] J;
+          node [shape=box,label="Services/Persistence/..."] D;
+
+          label = "Servlet Container";
+          color = black;
+      }
+      Client -> Network -> F -> J -> D;
+    }
+
+Exceptions before handing over to JAX-RS
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The servlet container usually offers limited options to handle errors in elegant ways for an API, which is why you want
+to avoid throwing exceptions there. However, it is possible: `see examples by Eugen Baeldung <https://www.baeldung.com/servlet-exceptions>`_
+
+You better avoid throwing exceptions from places like
+`Filter <https://jakarta.ee/specifications/platform/8/apidocs/javax/servlet/Filter.html>`_\ s or similar, as those
+require specialized code handling instead of going with the below JAX-RS exception handling.
+
+Exceptions after handing over to JAX-RS
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Cases 2) and 3) above are both happening in code easy to control in API code without fiddling with ``web.xml`` or similar.
+
+Easy to follow code examples and explanations:
+
+- https://www.baeldung.com/jersey-rest-api-with-spring#using-exceptionmapper (never mind this is about Spring, the technology is the same!)
+- https://www.baeldung.com/jersey-bean-validation#custom-exception-handler
+
+For Dataverse, all API exception handlers live in package `edu.harvard.iq.dataverse.api.errorhandlers`_.
+*Remember, this does not handle exceptions popping via web UI, thus not served by Jersey JAX-RS!*
+
+1. ``ThrowableHandler`` catches any Java exception thrown and uncatched in business logic (either by accident or on purpose)
```

Comment:
really minor tweak: `uncatched` might better be expressed as `uncaught` or `not caught`